core/src/main/scala/pickling/PBuilderReader.scala

```diff
@@ -19,6 +19,10 @@ trait Hintable {
 trait PBuilder extends Hintable {
   def beginEntry(picklee: Any): PBuilder
   def putField(name: String, pickler: PBuilder => Unit): PBuilder
+  def putDynamicFieldNames(names: List[String])(implicit namesPickler: SPickler[List[String]]): PBuilder =
```

Comment:
In fact, this method should document that after this call, all "putField" calls need the "name" to be in the same order passed in via "putDynamicFieldNames"